### ISSUE TITLE
[Pytorch] wrong computation of softmin 

### ISSUE THREAD

**Issue by arogozhnikov** [pytorch/pytorch#10669]
## Code example

```python
torch.nn.functional.softmin(torch.arange(2).float(), dim=0)
output tensor([-0.2689, -0.7311])
```

Error is in priorities (minus applied after softmax):
```
if dim is None:
        dim = _get_softmax_dim('softmin', input.dim(), _stacklevel)
return -input.softmax(dim) # minus here comes after softmax
```

## System Info
```
Collecting environment information...
PyTorch version: 0.4.1
Is debug build: No
CUDA used to build PyTorch: 9.0.176

OS: Ubuntu 16.04.3 LTS
GCC version: (Ubuntu 5.4.0-6ubuntu1~16.04.5) 5.4.0 20160609
CMake version: version 3.5.1

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: 9.0.176
GPU models and configuration: GPU 0: GeForce GTX 1070
Nvidia driver version: 396.24.02
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.7.0.4
/usr/lib/x86_64-linux-gnu/libcudnn_static_v7.a

Versions of relevant libraries:
[pip3] numpy (1.14.0)
[pip3] torch (0.4.1)
[pip3] torchvision (0.2.1)
[conda] Could not collect
```

**Comment by SsnL** [comment]
fixed on master. sorry about the bug.
